pkg/apis/settings/settings.go

```diff
@@ -92,15 +97,18 @@ func (*Settings) Inject(ctx context.Context, cm *v1.ConfigMap) (context.Context,
 	return ToContext(ctx, s), nil
 }
 
+func (*Settings) FromContext(ctx context.Context) coresettings.Injectable {
+	return FromContext(ctx)
+}
+
 func ToContext(ctx context.Context, s *Settings) context.Context {
 	return context.WithValue(ctx, ContextKey, s)
 }
 
 func FromContext(ctx context.Context) *Settings {
 	data := ctx.Value(ContextKey)
 	if data == nil {
-		// This is developer error if this happens, so we should panic
```

Comment:
Why change this?

Reply:
I think its hidden since I've squashed a lot of commits into 1 but this was `return nil` for most of the lifetime of this PR. I'll restore the original panic message / comment.